verifiers/internal/gha/verifier.go

```diff
@@ -147,13 +144,22 @@ func (v *GHAVerifier) VerifyImage(ctx context.Context,
 			fmt.Fprintf(os.Stderr, "unexpected error getting certificate from OCI registry %s", err)
 			continue
 		}
-		verifiedProvenance, builderID, verifyErr = verifyEnvAndCert(env,
+		verifiedProvenance, builderID, err = verifyEnvAndCert(env,
 			cert, provenanceOpts, builderOpts,
 			defaultContainerTrustedReusableWorkflows)
-		if verifyErr == nil {
+		if err == nil {
 			return verifiedProvenance, builderID, nil
 		}
+		errs = append(errs, err)
 	}
 
-	return nil, nil, fmt.Errorf("no valid attestations found on OCI registry: %w", verifyErr)
+	// Return the first error.
+	if len(errs) > 0 {
```

Comment:
Why do we need this check? If we reach here this means we did not return any provenance, so there must be `errs`?

Reply:
if there is no attestation, the loop will not enter errs will be nil. I don't know the exact behavior `RunCosignImageVerification`. If it fails when no attestations are found, I agree with you. Shall I keep the code or add a comment that atts is never empty?

Reply:
It is correct that cosign will fail when no attestations are there, but I think we should not rely on that. I overlooked that missing attestations would not enter the loop. Thank you!